code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -53,6 +52,17 @@
 		health = 100
 		maxHealth = 100
 
+/mob/living/simple_animal/bot/floorbot/Exited(atom/movable/A, atom/newloc)
+	if(A == tilestack)
+		if(tilestack && tilestack.max_amount < tilestack.amount) //split the stack if it exceeds its normal max_amount
+			var/iterations = round(tilestack.amount/tilestack.max_amount) //round() without second arg floors the value
+			for(var/a in 1 to iterations)
+				if(a == iterations)
+					tilestack.split_stack(null, tilestack.amount - tilestack.max_amount)
+				else
+					tilestack.split_stack(null, tilestack.max_amount)
```

Comment:
Does this code exist elsewhere? feels like it should

Reply:
No, it doesn't afaik.

Reply:
This should probably be on the stack or tile type and just called here.